ci/infra/azure/worker-instance.tf

```diff
@@ -3,28 +3,32 @@ resource "azurerm_network_interface" "worker" {
   name                = "${var.stack_name}-worker-${count.index}-nic"
   location            = azurerm_resource_group.resource_group.location
   resource_group_name = azurerm_resource_group.resource_group.name
+  depends_on          = [azurerm_subnet.nodes]
 
   ip_configuration {
     name                          = "internal"
     subnet_id                     = azurerm_subnet.nodes.id
     private_ip_address_allocation = "Dynamic"
+    public_ip_address_id          = var.create_bastionhost ? null : element(azurerm_public_ip.worker.*.id, count.index)
   }
 }
 
 resource "azurerm_network_interface_security_group_association" "worker" {
   count                     = var.workers
   network_interface_id      = element(azurerm_network_interface.worker.*.id, count.index)
   network_security_group_id = azurerm_network_security_group.worker.id
+  depends_on                = [azurerm_network_interface.worker, azurerm_network_security_group.worker, azurerm_linux_virtual_machine.worker]
```

Comment:
same here

Reply:
We only have template on v5 beta, therefore this PR will update it.